providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/sync/SyncStreamConnector.java

```diff
@@ -109,11 +98,12 @@ void observeEventStream(final BlockingQueue<QueuePayload> writeTo, final AtomicB
             log.debug("Initializing sync stream request");
             final GetMetadataRequest.Builder metadataRequest = GetMetadataRequest.newBuilder();
             GetMetadataResponse metadataResponse = GetMetadataResponse.getDefaultInstance();
+            restart();
```

Comment:
The main issue this PR fixes is that there was no call to restart the stream in this while loop, and error messages are final (meaning the stream needs to be restored).

This restart does that, instead of only starting the stream once it starts it whenever we break out of the inner loop here.